nextcord/interactions/application.py

```diff
@@ -0,0 +1,177 @@
+        self.name: str = data["name"]
+
+        try:
+            self.focused = data["focused"]
+        except KeyError:
+            self.focused = False
+
+    def __repr__(self) -> str:
+        return str(self.value)
+
+
+class ApplicationCommandInteraction(Interaction):
+    """Represents the interaction for all application commands.
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two interactions are equal.
+
+        .. describe:: x != y
+
+            Checks if two interactions are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the interaction's hash.
+
+    Attributes
+    ----------
+    application_command: Union[SlashApplicationSubcommand, BaseApplicationCommand]
+        The application command that triggered the interaction.
+    app_command_name: :class:`str`
+        The name of the application command that triggered the interaction.
+    app_command_id: :class:`int`
+        The application command ID that triggered the interaction.
+    options: List[SlashOptionData]
+        The application command options that have been given a value.
+    """
+
+    __slots__: Tuple[str, ...] = (
+        "application_command",
+        "app_command_name",
+        "app_command_id",
+        "options",
+    )
+
+    def __init__(self, *, data: InteractionPayload, state: ConnectionState) -> None:
+        super().__init__(data=data, state=state)
+
+        self.application_command: Optional[
+            Union[SlashApplicationSubcommand, BaseApplicationCommand]
+        ] = None
+
+    def _from_data(self, data: InteractionPayload) -> None:
+        super()._from_data(data=data)
+
+        self.app_command_name: str = self.data["name"]  # type: ignore # self.data should be present here
+        self.app_command_id: int = self.data["id"]  # type: ignore # self.data should be present here
+
+        try:
+            self.options: List[SlashOptionData] = self._get_application_options(self.data["options"])  # type: ignore # Data should be defined here
+        except KeyError:
+            self.options: List[SlashOptionData] = []
+
+    def _set_application_command(
+        self, app_cmd: Union[SlashApplicationSubcommand, BaseApplicationCommand]
+    ) -> None:
+        self.application_command = app_cmd
+
+    def _get_application_options(self, data):
+        options = data
+
+        if len(options) == 0:
+            # return empty list if no options exist
+            return []
+
+        # iterate through options to get inputs
+        while "options" in options[0]:
+            options = options[0]["options"]
+
+            if len(options) == 0:
+                # return empty list if no options exist
+                return []
+
+        # If we are here, then the user provided an input for some options
+        options_collection: List[SlashOptionData] = []
+
+        for option in options:
+            options_collection.append(SlashOptionData(option))
```

Comment:
There's an easy way to get around the circular import issues if this is what `SlashOptionData` is being used for in the code. Just import whatever you need in the scope of the function. 
For example:

```python
def foo():
    from .bar import soap
    soap()
```

And for any time `SlashOptionData` is being used for type hints outside the scope of a function, import it under a `TYPE_CHECKING` block and enable `from __future__ import annotations`.

Reply:
I assume `foo()` represents the `_get_application_options()` method in this case? It is customary to have all imports at the top of a python file. Is it really a good idea to import within a function mid file? 

(Yes, this is why the class is in this file- was originally going to put it in `nextcord.application_command.py`. Since you asked about this in the other comment, and the discussion will continue here, I shall resolve that conversation)